geth/jail/execution_policy.go

```diff
@@ -150,18 +141,16 @@ func postProcessRequest(vm *otto.Otto, req common.RPCCall, messageID string) {
 }
 
 // currentMessageID looks for `status.message_id` variable in current JS context
-func currentMessageID(ctx otto.Context) string {
-	if statusObj, ok := ctx.Symbols["status"]; ok {
-		messageID, err := statusObj.Object().Get("message_id")
-		if err != nil {
-			return ""
-		}
-		if messageID, err := messageID.ToString(); err == nil {
-			return messageID
-		}
+func currentMessageID(vm *vm.VM) string {
+	msgID, err := vm.Run("status.message_id")
+	if err != nil {
+		return ""
 	}
-
-	return ""
+	msgIDStr, err := msgID.ToString()
```

Comment:
There is also a handy `(*otto.Value).String()` method which returns empty string if there is an error.